fabric-fluid-v1/src/main/java/net/fabricmc/fabric/api/fluids/v1/container/FluidContainer.java

```diff
@@ -0,0 +1,82 @@
+package net.fabricmc.fabric.api.fluids.v1.container;
+
+import java.util.Iterator;
+
+import net.minecraft.fluid.Fluid;
+import net.minecraft.fluid.Fluids;
+
+import net.fabricmc.fabric.Action;
+import net.fabricmc.fabric.api.fluids.v1.container.volume.FluidVolume;
+
+/**
+ * An object that stores fluid, it is assumed that all fluid containers that only hold one fluid fluid Volumes.
+ */
+public interface FluidContainer extends Iterable<FluidVolume> {
```

Comment:
why not be a Collection, this is quite weird imo, I might be wrong

Reply:
I just used the most generic interface, I personally find it convenient

Reply:
extending AbstractCollection in the impls is overkill imo.